channeld/channeld.c

```diff
@@ -138,6 +139,8 @@ struct peer {
 	u32 fee_per_satoshi;
 
 	/* The scriptpubkey to use for shutting down. */
+	u32 *final_index;
```

Comment:
why pointers instead of direct containment?

Reply:
It's the optional thing, NULL means not there

Reply:
see related comment below